cparser/Elab.ml

```diff
@@ -2284,7 +2343,7 @@ let rec elab_stmt env ctx s =
       let a',env = elab_expr ctx.ctx_vararg loc env a in
       begin match Ceval.integer_expr env a' with
         | None ->
-            error loc "argument of 'case' must be an integer compile-time constant"
+            error loc "expression is not an integer constant expression"
```

Comment:
Best of both messages: `argument of `case` is not an integer constant expression`